ts/features/wallet/onboarding/bancomat/screens/SearchBankScreen.tsx

```diff
@@ -1,18 +1,145 @@
-import { View } from "native-base";
+import { Content, Input, Item, Label, Text, View } from "native-base";
 import * as React from "react";
+import { FlatList, ListRenderItemInfo, SafeAreaView } from "react-native";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
+import { debounce } from "lodash";
+import { Link } from "../../../../../components/core/typography/Link";
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { GlobalState } from "../../../../../store/reducers/types";
+import IconFont from "../../../../../components/ui/IconFont";
+import { BankPreviewItem } from "../components/BankPreviewItems";
+import { abis } from "../mock/mockData";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import I18n from "../../../../../i18n";
+import { navigateBack } from "../../../../../store/actions/navigation";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { withLightModalContext } from "../../../../../components/helpers/withLightModalContext";
+import { LightModalContextInterface } from "../../../../../components/ui/LightModal";
+import TosBonusComponent from "../../../../bonus/bonusVacanze/components/TosBonusComponent";
+import { sortAbiByName } from "../utils/abi";
+import { Body } from "../../../../../components/core/typography/Body";
 
+type Props = LightModalContextInterface &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
 /**
  * This screen allows the user to choose a specific bank to search for their Bancomat.
  * the user can also choose not to specify any bank and search for all Bancomat in his name
  * @constructor
  */
-export const SearchBankScreen: React.FunctionComponent = () => <View />;
+const SearchBankScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState(
+    [] as typeof props.bankList
+  );
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
 
-const mapStateToProps = (_: GlobalState) => ({});
+  const renderListItem = (isList: boolean) => (
+    info: ListRenderItemInfo<any>
+  ) => (
+    <BankPreviewItem
+      bank={info.item}
+      inList={isList}
+      onPress={props.onItemPress}
+    />
+  );
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBankScreen);
+  const debounceSearch = debounce((text: string) => {
```

Comment:
`debounce` doesn't work with anonymous functions. This implementation runs a search foreach typed character

This is an implementation that executes a search only after delay milliseconds

```typescript
  const performSearch = (text: string) => {
    if (text.length === 0) {
      setFilteredList([]);
      setIsSearchStarted(false);
      return;
    }
    const resultList = props.bankList.filter(
      bank =>
        bank.abi.toLowerCase().indexOf(text.toLowerCase()) > -1 ||
        bank.name.toLowerCase().indexOf(text.toLowerCase()) > -1
    );
    setFilteredList(sortAbiByName(resultList));
    setIsSearchStarted(false);
  };
  const debounceRef = React.useRef(debounce(performSearch, delay));

  const [isSearchStarted, setIsSearchStarted] = React.useState(false);

  React.useEffect(() => {
    setIsSearchStarted(true);
    debounceRef.current(searchText);
  }, [searchText]);

  const handleFilter = (text: string) => {
    setSearchText(text);
  };
```